From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ghosuto <clash.raja10@gmail.com>
Date: Thu, 9 Jul 2026 08:23:17 +0000
Subject: [PATCH 213/220] SystemUI: Gate WindowRootViewBinder logs behind DEBUG
 flag

[MikolajQ: Ported to Android 17: the blur plumbing moved to BlurChoreographer, the remaining binder logs are gated]

Signed-off-by: Ghosuto <clash.raja10@gmail.com>
---
 .../systemui/window/ui/WindowRootViewBinder.kt        | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/window/ui/WindowRootViewBinder.kt b/packages/SystemUI/src/com/android/systemui/window/ui/WindowRootViewBinder.kt
index dd351a95e..ffafeaa8b 100644
--- a/packages/SystemUI/src/com/android/systemui/window/ui/WindowRootViewBinder.kt
+++ b/packages/SystemUI/src/com/android/systemui/window/ui/WindowRootViewBinder.kt
@@ -37,6 +37,7 @@ import kotlinx.coroutines.flow.filter
  */
 object WindowRootViewBinder {
     private const val TAG = "WindowRootViewBinder"
+    private val DEBUG = Log.isLoggable(TAG, Log.DEBUG)
 
     fun bind(
         view: WindowRootView,
@@ -48,7 +49,7 @@ object WindowRootViewBinder {
         if (SceneContainerFlag.isEnabled) return
 
         view.repeatWhenAttached(mainDispatcher) {
-            Log.d(TAG, "Binding root view")
+            if (DEBUG) Log.d(TAG, "Binding root view")
             view.viewModel(
                 minWindowLifecycleState = WindowLifecycleState.ATTACHED,
                 factory = { viewModelFactory.create() },
@@ -59,7 +60,9 @@ object WindowRootViewBinder {
                 }
                 blurChoreographer.registerOnBlurAppliedListener(onBlurApplied)
                 try {
-                    Log.d(TAG, "Launching coroutines that update window root view state")
+                    if (DEBUG) {
+                        Log.d(TAG, "Launching coroutines that update window root view state")
+                    }
                     launchTraced("early-wakeup") {
                         viewModel.isPersistentEarlyWakeupRequired.collect { wakeupRequired ->
                             blurChoreographer.setPersistentEarlyWakeup(wakeupRequired)
@@ -76,7 +79,9 @@ object WindowRootViewBinder {
                     awaitCancellation()
                 } finally {
                     blurChoreographer.clearOnBlurAppliedListener()
-                    Log.d(TAG, "Wrapped up coroutines that update window root view state")
+                    if (DEBUG) {
+                        Log.d(TAG, "Wrapped up coroutines that update window root view state")
+                    }
                 }
             }
         }
